config/v1/types_oauth.go

```diff
@@ -583,3 +583,35 @@ type OAuthList struct {
 
 	Items []OAuth `json:"items"`
 }
+
+// OAuthSpec contains desired cluster auth configuration
+type OAuthSpec struct {
+	// identityProviders is an ordered list of ways for a user to identify themselves.
+	// When this list is empty, no identities are provisioned for users.
+	// +optional
+	IdentityProviders []IdentityProvider `json:"identityProviders,omitempty"`
+
+	// tokenConfig contains options for authorization and access tokens
+	TokenConfig TokenConfig `json:"tokenConfig"`
+
+	// templates allow you to customize pages like the login page.
+	// +optional
+	Templates OAuthTemplates `json:"templates"`
+
+	Audit OAuthAudit `json:"audit"`
```

Comment:
This is the field we would like to add to the struct on the top. We should consider adding some comments to it too. WDYT?